File: src/main/java/org/gangel/graphomance/TestCase.java
package org.gangel.graphomance;

public interface TestCase {

    void initialize(Connection conn, SessionProducer sessionProducer);

    void setUpTest();

    void createTestData();

    void performTest();

    void cleanUpData();

    void terminate();

    DbType[] skipFor();

    default boolean skipFor(DbType dbType) {
        DbType[] skipForTypes = skipFor();
        if (skipForTypes == null) {
            return false;
        }
        for(DbType t : skipForTypes) {
            if (t == dbType) {
                return true;
            }
        }
        return false;
    }

}

File: src/main/kotlin/org/gangel/graphomance/TestCase.kt
package org.gangel.graphomance

interface TestCase {
	fun setUpTest(session: Session)
	fun createTestData(session: Session)
	fun performTest(session: Session)
	fun cleanUpData(session: Session)
	fun skipFor(): Array<DbType>?
	fun skipFor(dbType: DbType): Boolean {
		val skipForTypes = skipFor() ?: return false
		for (t in skipForTypes) {
			if (t == dbType) {
				return true
			}
		}
		return false
	}
}